sdk/provisioning/Azure.Provisioning/src/Primitives/ProvisionableResource.cs

```diff
@@ -33,7 +33,22 @@ public abstract class ProvisionableResource(string bicepIdentifier, ResourceType
     /// Gets whether this is referencing an existing resource or we're defining
     /// a new resource.
     /// </summary>
-    public bool IsExistingResource { get; protected set; } = false;
+    public bool IsExistingResource
+    {
+        get => _isExisting;
+        protected set
+        {
+            _isExisting = value;
+            if (_isExisting)
```

Comment:
Possible threading bug?
```suggestion
            _isExisting = value;
            if (value)
```

Reply:
None of this is designed to be thread safe.  You'll have to manage synchronization yourself if you want to share these objects across threads (i.e., just like our Azure SDK model types).